### PR TITLE
Update GoScaleio Version

### PR DESCRIPTION
**Pull request by doriac11**
# Description
- Update goscaleio
- Fix documentation issue
- Update Yaml Dependency for Security Risk (Had to comment out the terraform-plugin-docs to fix this)
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

### CODE REVIEW

**Review by nikitajoshi1 — APPROVED** [review]

**Review by shekhar-j — APPROVED** [review]

File: templates/resources/compatibility_management.md.tmpl [hunk 1]
```diff
@@ -26,26 +26,6 @@ description: |-
 
 {{ .Description | trimspace }}
 
-**Please consider the following points before using cluster resource.**
-
-1. For PowerFlex 4.x, the PowerFlex Manager must be installed as a prerequisite. The required packages should be uploaded to the PowerFlex Manager.
-
-2. For PowerFlex 3.x, a Gateway server is a prerequisite. The required packages should be uploaded to the gateway. The Package resource can be used for uploading packages to the gateway.
-
```
Comment (rounak-adhikary) [inline templates/resources/compatibility_management.md.tmpl:34]: are these warning removals expected?

Comment (doriac11) [reply templates/resources/compatibility_management.md.tmpl:34]: Yea they were accidentally added when creating this template, they were for cluster.

**Comment by doriac11** [comment]
run e2e test

**Review by shekhar-j — APPROVED** [review]

**Comment by doriac11** [comment]
run e2e test

**Comment by doriac11** [comment]
e2e test all passed except for this which is an environment issue
```
=== RUN   TestAccDatasourceSdc
    sdc_datasource_test.go:50: Step 1/3 error: Check failed: Check 1/1 error: data.powerflex_sdc.selected: Attribute 'sdcs.0.sdc_ip' expected "", got "10.247.101.248"
--- FAIL: TestAccDatasourceSdc 
```

**Review by shekhar-j — APPROVED** [review]

**Review by nikitajoshi1 — APPROVED** [review]

**Pull request merged**